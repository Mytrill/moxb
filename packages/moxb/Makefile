all:
	cd ../.. && $(MAKE)

# this will forward any other make goal to the parent makefile
$(MAKECMDGOALS):
	cd ../.. && $(MAKE) $(MAKECMDGOALS)
