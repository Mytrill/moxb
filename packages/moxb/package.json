{
  "name": "@moxb/moxb",
  "version": "0.1.3",
  "description": "MobX data binding",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "main": "build",
  "typings": "build/",
  "files": [
    "build/**/*"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/moxb/moxb.git"
  },
  "author": "moxb (moxb.org)",
  "license": "MIT",
  "dependencies": {
    "tslib": "^1.9.3"
  },
  "peerDependencies": {
    "mobx": "^4.3.0 || ^5.0.0"
  },
  "devDependencies": {
    "@types/jest": "^23.1.5",
    "@types/meteor": "^1.4.16",
    "jest": "^23.3.0",
    "prettier": "^1.13.7",
    "ts-jest": "^23.0.0",
    "typescript": "^2.9.2"
  },
  "bugs": {
    "url": "https://github.com/moxb/moxb/issues"
  },
  "homepage": "https://github.com/moxb/moxb#readme",
  "keywords": [
    "mobx",
    "UI"
  ]
}
