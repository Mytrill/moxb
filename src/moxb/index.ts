export * from './impl';
export * from './meteor';
export * from './types';
