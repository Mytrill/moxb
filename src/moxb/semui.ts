export * from './action/ActionUi';
export * from './bind/BindUi';
export * from './bool/BoolUi';
export * from './confirm/ConfirmUi';
export * from './many-of/ManyOfUi';
export * from './modal/ModalUi';
export * from './numeric/NumericUi';
export * from './one-of/OneOfUi';
export * from './table/TablePaginationUi';
export * from './table/TableSearchUi';
export * from './table/TableUi';
export * from './text/TextUi';
