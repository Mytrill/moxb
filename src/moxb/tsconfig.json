{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES5",
        "jsx": "react",
        "lib": ["DOM","ES6","ES2015","ES2017", "DOM.Iterable","ScriptHost"],
        "importHelpers":true,
        "experimentalDecorators": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strict": true,
        "outDir": "../build",
        "declaration": true,
        "skipLibCheck": true,
        "strictNullChecks": true,
        "removeComments": false
    },
    "compilerOptions": {
        "baseUrl": "",
        "paths": {
            "*": [
                "*",
                "../../optional-modules/node_modules/*"
            ]
        }
    },
    "exclude": [
        "../node_modules"
    ]
}
