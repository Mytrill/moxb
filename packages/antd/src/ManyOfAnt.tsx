import { toJS } from 'mobx';
import { observer } from 'mobx-react';
import * as React from 'react';
import { Form, Select } from 'antd';
import { BindAntFormProps, BindAntProps, labelWithHelp, parseProps } from './BindAnt';
import { ManyOf } from '@moxb/moxb';
import { SelectProps } from 'antd/lib/select';

@observer
export class ManyOfAnt extends React.Component<BindAntProps<ManyOf> & SelectProps> {
    render() {
        const { operation, invisible, mode, children, defaultValue, label, help, placeholder, ...props } = parseProps(
            this.props,
            this.props.operation
        );
        if (invisible || operation.invisible) {
            return null;
        }
        // make sure the value is not a mobx object...
        const value = toJS(operation.value);
        return (
            <Select
                onChange={(selectionValue: any) => operation.setValue(selectionValue)}
                value={value}
                placeholder={placeholder}
                defaultValue={defaultValue}
                mode={typeof mode === 'undefined' ? 'default' : mode}
                {...props}
            >
                {operation.choices.map(opt => (
                    <Select.Option key={opt.value} value={opt.value}>
                        {opt.label}
                    </Select.Option>
                ))}
                {children}
            </Select>
        );
    }
}

@observer
export class ManyOfFormAnt extends React.Component<BindAntFormProps & SelectProps & BindAntProps<ManyOf>> {
    render() {
        const { operation, label, help, invisible, formStyle, ...props } = parseProps(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        return (
            <Form.Item label={label} help={help} style={formStyle || undefined}>
                <ManyOfAnt operation={operation} {...props as any} />
            </Form.Item>
        );
    }
}
