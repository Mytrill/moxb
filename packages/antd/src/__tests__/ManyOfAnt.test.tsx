import { ManyOfImpl } from '@moxb/moxb';
import { shallow } from 'enzyme';
import * as React from 'react';
import { ManyOfAnt, ManyOfCheckboxAnt, ManyOfFormAnt } from '../ManyOfAnt';

const manyChoices: any[] = [];
const allChoices = [{ label: 'Banana', value: 'b' }, { label: 'Apples', value: 'a' }, { label: 'Peaches', value: 'p' }];

describe('ManyOfAnt', function() {
    it('should render a select control by default', function() {
        const operation = new ManyOfImpl({
            id: 'Id.manyOf',
            label: 'Choose your snack',
            choices: () => allChoices,
            initialValue: () => manyChoices,
            placeholder: () => 'Please select',
            onSave: () => {},
        });
        expect(shallow(<ManyOfAnt operation={operation} />)).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const operation = new ManyOfImpl({
            id: 'Id.manyOf',
            label: 'Choose your snack',
            choices: () => allChoices,
            initialValue: () => manyChoices,
            placeholder: () => 'Please select',
            onSave: () => {},
            invisible: () => true,
        });
        expect(shallow(<ManyOfAnt operation={operation} />).type()).toBeNull();
    });
});

describe('ManyOfCheckboxAnt', function() {
    it('should render a checkboxes by default', function() {
        const operation = new ManyOfImpl({
            id: 'Id.manyOf',
            label: 'Choose your snack',
            choices: () => allChoices,
            initialValue: () => manyChoices,
            placeholder: () => 'Please select',
            onSave: () => {},
        });
        expect(shallow(<ManyOfCheckboxAnt operation={operation} />)).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const operation = new ManyOfImpl({
            id: 'Id.manyOf',
            label: 'Choose your snack',
            choices: () => allChoices,
            initialValue: () => manyChoices,
            placeholder: () => 'Please select',
            onSave: () => {},
            invisible: () => true,
        });
        expect(shallow(<ManyOfCheckboxAnt operation={operation} />).type()).toBeNull();
    });
});

describe('ManyOfFormAnt', function() {
    it('should render a checkboxes by default', function() {
        const operation = new ManyOfImpl({
            id: 'Id.manyOf',
            label: 'Choose your snack',
            choices: () => allChoices,
            initialValue: () => manyChoices,
            placeholder: () => 'Please select',
            onSave: () => {},
        });
        expect(shallow(<ManyOfFormAnt operation={operation} />)).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const operation = new ManyOfImpl({
            id: 'Id.manyOf',
            label: 'Choose your snack',
            choices: () => allChoices,
            initialValue: () => manyChoices,
            placeholder: () => 'Please select',
            onSave: () => {},
            invisible: () => true,
        });
        expect(shallow(<ManyOfFormAnt operation={operation} />).type()).toBeNull();
    });
});
