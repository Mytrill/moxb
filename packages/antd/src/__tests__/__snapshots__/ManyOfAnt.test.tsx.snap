// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManyOfAnt should render a select control by default 1`] = `
<Select
  choiceTransitionName="zoom"
  disabled={false}
  id="id-many_of"
  label="Choose your snack"
  mode="default"
  onChange={[Function]}
  prefixCls="ant-select"
  readOnly={false}
  showSearch={false}
  transitionName="slide-up"
  value={Array []}
>
  <Option
    key="b"
    value="b"
  >
    Banana
  </Option>
  <Option
    key="a"
    value="a"
  >
    Apples
  </Option>
  <Option
    key="p"
    value="p"
  >
    Peaches
  </Option>
</Select>
`;

exports[`ManyOfCheckboxAnt should render a checkboxes by default 1`] = `
<CheckboxGroup
  disabled={false}
  id="id-many_of"
  label="Choose your snack"
  onChange={[Function]}
  options={Array []}
  prefixCls="ant-checkbox"
  readOnly={false}
  value={Array []}
>
  <Row
    gutter={0}
  >
    <Col
      key="b"
    >
      <Checkbox
        indeterminate={false}
        prefixCls="ant-checkbox"
        value="b"
      >
        Banana
      </Checkbox>
    </Col>
    <Col
      key="a"
    >
      <Checkbox
        indeterminate={false}
        prefixCls="ant-checkbox"
        value="a"
      >
        Apples
      </Checkbox>
    </Col>
    <Col
      key="p"
    >
      <Checkbox
        indeterminate={false}
        prefixCls="ant-checkbox"
        value="p"
      >
        Peaches
      </Checkbox>
    </Col>
  </Row>
</CheckboxGroup>
`;

exports[`ManyOfFormAnt should render a checkboxes by default 1`] = `
<FormItemAnt
  operation={
    bound ManyOfImpl {
      "_errors": Array [],
      "_isSaving": false,
      "_value": undefined,
      "constructor": [Function],
      "doClearErrors": [Function],
      "doGetValue": [Function],
      "doOnValidate": [Function],
      "doPlaceholder": [Function],
      "doSetValue": [Function],
      "domId": "id-many_of",
      "getDisabled": [Function],
      "getErrors": [Function],
      "getHelp": [Function],
      "getInitialValue": [Function],
      "getInvisible": [Function],
      "getLabel": [Function],
      "getReadonly": [Function],
      "getReason": [Function],
      "id": "Id.manyOf",
      "impl": Object {
        "choices": [Function],
        "id": "Id.manyOf",
        "initialValue": [Function],
        "label": "Choose your snack",
        "onSave": [Function],
        "placeholder": [Function],
      },
      "setError": [Function],
      Symbol(mobx did run lazy initializers): true,
      Symbol(mobx administration): ObservableObjectAdministration$$1 {
        "defaultEnhancer": [Function],
        "keysAtom": Atom$$1 {
          "diffValue": 0,
          "isBeingObserved": false,
          "isPendingUnobservation": false,
          "lastAccessedBy": 0,
          "lowestObserverState": 2,
          "name": "ManyOfImpl@73.keys",
          "observers": Set {},
        },
        "name": "ManyOfImpl@73",
        "target": [Circular],
        "values": Map {
          "_errors" => Array [],
          "label" => "Choose your snack",
          "disabled" => false,
          "enabled" => true,
          "reason" => undefined,
          "invisible" => false,
          "readOnly" => false,
          "help" => undefined,
          "errors" => Array [],
          "error" => undefined,
          "hasErrors" => false,
          "_value" => undefined,
          "_isSaving" => false,
          "inputType" => undefined,
          "value" => Array [],
          "placeholder" => "Please select",
          "required" => undefined,
          "isInitialValue" => true,
          "choices" => Array [
            Object {
              "label": "Banana",
              "value": "b",
            },
            Object {
              "label": "Apples",
              "value": "a",
            },
            Object {
              "label": "Peaches",
              "value": "p",
            },
          ],
        },
      },
    }
  }
>
  <ManyOfAnt
    disabled={false}
    id="id-many_of"
    operation={
      bound ManyOfImpl {
        "_errors": Array [],
        "_isSaving": false,
        "_value": undefined,
        "constructor": [Function],
        "doClearErrors": [Function],
        "doGetValue": [Function],
        "doOnValidate": [Function],
        "doPlaceholder": [Function],
        "doSetValue": [Function],
        "domId": "id-many_of",
        "getDisabled": [Function],
        "getErrors": [Function],
        "getHelp": [Function],
        "getInitialValue": [Function],
        "getInvisible": [Function],
        "getLabel": [Function],
        "getReadonly": [Function],
        "getReason": [Function],
        "id": "Id.manyOf",
        "impl": Object {
          "choices": [Function],
          "id": "Id.manyOf",
          "initialValue": [Function],
          "label": "Choose your snack",
          "onSave": [Function],
          "placeholder": [Function],
        },
        "setError": [Function],
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "ManyOfImpl@73.keys",
            "observers": Set {},
          },
          "name": "ManyOfImpl@73",
          "target": [Circular],
          "values": Map {
            "_errors" => Array [],
            "label" => "Choose your snack",
            "disabled" => false,
            "enabled" => true,
            "reason" => undefined,
            "invisible" => false,
            "readOnly" => false,
            "help" => undefined,
            "errors" => Array [],
            "error" => undefined,
            "hasErrors" => false,
            "_value" => undefined,
            "_isSaving" => false,
            "inputType" => undefined,
            "value" => Array [],
            "placeholder" => "Please select",
            "required" => undefined,
            "isInitialValue" => true,
            "choices" => Array [
              Object {
                "label": "Banana",
                "value": "b",
              },
              Object {
                "label": "Apples",
                "value": "a",
              },
              Object {
                "label": "Peaches",
                "value": "p",
              },
            ],
          },
        },
      }
    }
    readOnly={false}
  />
</FormItemAnt>
`;
